File: SumOfNaturalNumber.cpp
#include<stdio.h>
#include<iostream>
#include<stdlib.h>

using namespace std;

int sum(int n){
   if(n == 0){
    return 0;
   }else{
    return sum(n-1)+n;
   }
};

int Isum(int n){
    int s = 0;
    int i;
    for(i=1; i<= n; i++){
        s = s+i;
    }
    return s;
};

int main()
{   
    //out 28
    int r;
    r = sum(5);
    cout<< r <<endl;
    return 0;
}

File: SumOfNaturalNumber.php
<?php


function sum($n){
    if($n == 0){
        return 0;
    }else{
        return sum($n-1) + $n;
    }
}

function Isum($n){
    $s = 0;
    for($i=1; $i <= $n; $i++){
        $s = $s + $i;
    }
    return $s;
}

$r = Isum(5);
echo $r . "\n";

?>